src/MySqlConnector/MySqlDecimal.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace MySqlConnector;
+
+public readonly struct MySqlDecimal
+{
+	private readonly string value;
+
+	internal MySqlDecimal(string val)
+	{
+		string regexWithDecimal = @"^-?([1-9][0-9]*|0)(\.[0-9]+)?$";
+		string regexWithOutDecimal = @"^-?([1-9][0-9]*|0)$";
+		var matchWithDecimal = Regex.Match(val, regexWithDecimal, RegexOptions.IgnoreCase);
+		var matchWithOutDecimal = Regex.Match(val, regexWithOutDecimal, RegexOptions.IgnoreCase);
```

Comment:
These could be `static readonly Regex` fields to avoid depending on the global regex cache here. I also think they could be combined into one regex.

These regexes allow `-0` which arguably shouldn't be allowed (as MySQL doesn't support the concept of negative zero AFAIK).

Reply:
right , regexWithOutDecimal is updated to take care of this bug here